dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/k8s/impl/K8sYamlTaskExecutor.java

```diff
@@ -0,0 +1,228 @@
+            parseLogOutput();
+
+            registerBatchWatcher(Integer.toString(taskInstanceId), result);
+
+            if (podLogOutputFuture != null) {
+                try {
+                    // Wait kubernetes pod log collection finished
+                    podLogOutputFuture.get();
+                } catch (ExecutionException e) {
+                    log.error("Handle pod log error", e);
+                }
+            }
+        } catch (Exception e) {
+            cancelApplication(yamlContentString);
+            Thread.currentThread().interrupt();
+            result.setExitStatusCode(EXIT_CODE_FAILURE);
+            throw e;
+        }
+        return result;
+    }
+
+
+    @Override
+    public void cancelApplication(String k8sParameterStr) {
+        if (metadata != null) {
+            stopJobOnK8s(k8sParameterStr);
+        }
+    }
+
+    @Override
+    public void submitJob2k8s(String yamlContentString) {
+        int taskInstanceId = taskRequest.getTaskInstanceId();
+        String taskName = taskRequest.getTaskName().toLowerCase(Locale.ROOT);
+        YamlContent yamlContent = JSONUtils.parseObject(yamlContentString, YamlContent.class);
+        metadata = abstractK8sOperation.buildMetadata(yamlContent);
+        Map<String, String> labelMap = metadata.getMetadata().getLabels();
+        String k8sJobName = String.format("%s-%s", taskName, taskInstanceId);
+        if (MapUtils.isEmpty(labelMap)) {
+            labelMap = new HashMap<String, String>(1);
+        }
+        // add special label which make people to get it simple
+        labelMap.put(LAYER_LABEL, LAYER_LABEL_VALUE);
+        labelMap.put(NAME_LABEL, k8sJobName);
+        try {
+            log.info("[K8sYamlJobExecutor-{}-{}] start to submit job", taskName, taskInstanceId);
+            abstractK8sOperation.createOrReplaceMetadata(metadata);
+            log.info("[K8sYamlJobExecutor-{}-{}] submitted job successfully", taskName, taskInstanceId);
+        } catch (Exception e) {
+            log.error("[K8sYamlJobExecutor-{}-{}] fail to submit job", taskName, taskInstanceId);
+            throw new TaskException("K8sYamlJobExecutor fail to submit job", e);
+        }
+    }
+
+
+    @Override
+    public void stopJobOnK8s(String k8sParameterStr) {
+        K8sTaskMainParameters k8STaskMainParameters =
+                JSONUtils.parseObject(k8sParameterStr, K8sTaskMainParameters.class);
+        String namespaceName = k8STaskMainParameters.getNamespaceName();
+        String jobName = metadata.getMetadata().getName();
+        try {
+            if (Boolean.TRUE.equals(k8sUtils.jobExist(jobName, namespaceName))) {
+                k8sUtils.deleteJob(jobName, namespaceName);
+            }
+        } catch (Exception e) {
+            log.error("[K8sYamlJobExecutor-{}] fail to stop job", jobName);
+            throw new TaskException("K8sYamlJobExecutor fail to stop job", e);
+        }
+    }
+
+
+    void generateOperation(){
```

Comment:
May I ask why do we need to distinguish the types of yaml and process them differently? Why not apply a yaml directly, for security reasons?

Reply:
> May I ask why do we need to distinguish the types of yaml and process them differently? Why not apply a yaml directly, for security reasons?

The created api is different, and the way to obtain the status by re-monitoring after restarting, the monitoring method may also be different.